standardhealth/fsh-mcode
# Sushi 0.9.0 Updates

SUSHI 0.9.0 introduces features that allow us to switch to SUSHI-generation of the IG.  I've made the following changes to accommodate this:

- Added ig-data folder
- Moved ig.ini and package-list.json from build/ to ig-data/
- Moved build/input/images to ig-data/input/images
- Moved build/input/includes to ig-data/input/includes
- Moved build/input/pagecontent to ig-data/input/pagecontent
- Minor modifications to package.json (semantic versioning, fixed contact/maintainers)
- Minor modifications to package-list.json (semantic versioning, removed boilerplate text)
- Minor modifications to ig.ini (removed extraneous template)

Note that the cache files in `build/input-cache` are retained (except the publisher jar) to enable quick builds.

To test this:
- Ensure you have sushi 0.9.0 (`sushi -v`)
- Switch to this branch (`git checkout sushi-090-updates)
- Clean extraneous files in build (see below*)
- Run sushi from the `fsh-mcode` folder (e.g., `sushi .`)
- Build the IG via the normal means

The easiest way to clean extraneous files is to use `git clean`.  First you should do a dry run, which will tell you what files _would_ be deleted without actually deleting them:
```
git clean -xfdn
```
If you're comfortable with what will be deleted, you can do it for real:
```
git clean -xfd
```
